### PR TITLE
A4A: Port the Jetpack Manage StoredCreditCard component to A4A.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request ports the [StoredCreditCard](https://github.com/Automattic/wp-calypso/tree/trunk/client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2) component from Jetpack Manage to A4A. Most of the code is copy and paste, but there are also a few improvements to simplify some of the code.

Screenshot 2024-03-06 at 7 24 39 PM

Closes https://github.com/Automattic/jetpack-genesis/issues/278

## Proposed Changes

* Port the [StoredCreditCard](https://github.com/Automattic/wp-calypso/tree/trunk/client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2) component and its related components and hooks (e.g., dialog, confirmation, etc.).
* Mock mutation calls for both SetAsPrimaryCard and Delete operations.

## Testing Instructions

* Switch branch: `git checkout add/a4a-credit-card-component`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/purchases/payment-methods
* Confirm That the Credit cards listed work the same from https://cloud.jetpack.com/partner-portal/payment-methods
* Note that **SetAsPrimaryCard** and **Delete** operations are mocked.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by cleacos — APPROVED** [review]
It works as expected.

LGTM!

**Comment by jkguidaven** [comment]
Thanks for the review @cleacos 🙏

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/stored-credit-card-delete-dialog/delete-primary-confirmation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	if ( ! altCard ) {
+		return (
+			<div className="delete-primary-card-confirmation">
+				<div className="delete-primary-card-confirmation__card">
+					<p className="delete-primary-card-confirmation__notice">
+						{ translate(
+							'Issuing new licenses will be paused until you add a new primary payment method. ' +
+								'Additionally, the existing licenses will be revoked at the end of their respective terms.'
+						) }
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/stored-credit-card-delete-dialog/delete-primary-confirmation/index.tsx]: I'm curious about this `translate`  usage and if the translation system takes the whole text with the `+` concantation.

Comment (jkguidaven) [reply client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/stored-credit-card-delete-dialog/delete-primary-confirmation/index.tsx]: I will probably split this string into two `translate` calls. The linting was auto-fixing this due to hitting max-line. Thanks for bringing this to my attention.

Comment (jkguidaven) [reply client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/stored-credit-card-delete-dialog/delete-primary-confirmation/index.tsx]: I lied 😀; this was intentional from the original source file, and I assumed it was the linting auto-fixing. This has been fixed.

Comment (cleacos) [reply client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/stored-credit-card-delete-dialog/delete-primary-confirmation/index.tsx]: So the answer is yes, `translate` (and translators) manage this split well... Or that is what we guess :)

Comment (jkguidaven) [reply client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/stored-credit-card-delete-dialog/delete-primary-confirmation/index.tsx]: Yes, the system seems to recognize it. But for good practice(I think), this should be set into one string.

**Pull request merged**